apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -118,6 +118,37 @@ type APMFeatureConfig struct {
 	// Path Default: `/var/run/datadog/apm.socket`
 	// +optional
 	UnixDomainSocketConfig *UnixDomainSocketConfig `json:"unixDomainSocketConfig,omitempty"`
+
+	// Single Step Instrumentation allows the agent to inject the Datadog APM libraries into all pods in the cluster.
+	// Feature is in beta.
+	// See also: https://docs.datadoghq.com/tracing/trace_collection/single-step-apm
+	// Enabled Default: false
+	// +optional
+	SingleStepInstrumentation *SingleStepInstrumentation `json:"instrumentation,omitempty"`
+}
+
+// SingleStepInstrumentation contains the config for the namespaces to target and the library to inject.
+type SingleStepInstrumentation struct {
+	// Enabled enables injecting the Datadog APM libraries into all pods in the cluster.
+	// Default: false
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// EnabledNamespaces specifies the namespaces of the pods to inject the Datadog APM libraries into.
+	// +optional
+	// +listType=set
+	EnabledNamespaces []string `json:"enabledNamespaces,omitempty"`
+
+	// DisabledNamespaces specifies the namespaces of the pods to not inject the Datadog APM libraries into.
```

Comment:
```suggestion
	// DisabledNamespaces disables injecting the Datadog APM libraries into pods in specific namespaces.
```
(used the same wording as in the helm PR)